packages/binding-mbus/src/mbus-connection.ts

```diff
@@ -182,21 +182,25 @@ export class MBusConnection {
                 // inform all the operations that the connection cannot be recovered
                 this.queue.forEach((transaction) => {
                     transaction.operations.forEach((op) => {
-                        op.failed(error);
+                        op.failed(`${error}`);
                     });
                 });
             }
         } else if (this.connected && this.currentTransaction == null && this.queue.length > 0) {
             // take next transaction from queue and execute
             this.currentTransaction = this.queue.shift();
-            try {
-                await this.executeTransaction(this.currentTransaction);
-                this.currentTransaction = null;
-                this.trigger();
-            } catch (err) {
-                warn(`Transaction failed: ${err}`);
-                this.currentTransaction = null;
-                this.trigger();
+            if (!this.currentTransaction) {
+                warn(`Current transaction is undefined -> transaction not executed`);
+            } else {
+                try {
+                    await this.executeTransaction(this.currentTransaction);
+                    this.currentTransaction = undefined;
+                    this.trigger();
+                } catch (err) {
+                    warn(`Transaction failed: ${err}`);
+                    this.currentTransaction = undefined;
+                    this.trigger();
+                }
             }
```

Comment:
I guess the function could also return early here, right? That would get rid of the else clause and therefore another level of nesting.

```suggestion
            if (!this.currentTransaction) {
                warn(`Current transaction is undefined -> transaction not executed`);
                return;
            }

            try {
                await this.executeTransaction(this.currentTransaction);
                this.currentTransaction = undefined;
                this.trigger();
            } catch (err) {
                warn(`Transaction failed: ${err}`);
                this.currentTransaction = undefined;
                this.trigger();
            }
```

Reply:
Matter of taste but Yes, let's do it..

I grew up with MISRA that educated that _a function should have a single point of exit at the end_. 🙃

Reply:
Oh, yeah, I also did some research after you posted your initial comment and came across that page :D Interesting paradigm ;) I guess for JavaScript/TypeScript it might be a bit more idiomatic to use these kinds of "guard clauses", though, since we do not need to manually clean up memory, for example, before returning. In the end, it probably depends on the respective function, though, which approach is the most readable and maintainable one, as there are also many cases where one return statement at the end is the simplest approach.